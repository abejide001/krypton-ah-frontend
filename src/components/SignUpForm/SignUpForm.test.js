import React from 'react';
import { shallow, mount } from 'enzyme';
import { SignUp } from '.';
import RedirectLoader from '../Loader';

describe('SignUp Form', () => {
  let props = {
    signup: () => {}
  };
  test('should matches the snapshot', () => {
    const wrapper = shallow(<SignUp />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Simulate submit button ', () => {
    const wrapper = mount(<SignUp {...props} />);
    wrapper.find('form').simulate('submit', {
      preventDefault: jest.fn()
    });
    expect(wrapper.find('form')).toBeDefined();
  });

  it('should change the state after change the input value', () => {
    const newValue = 'iolan@anla.com';
    const wrapper = mount(<SignUp />);
    wrapper.find('input[type="email"]').simulate('change', {
      target: {
        name: newValue
      }
    });
    wrapper.update();
    expect(wrapper.find('input[type="email"]').prop('name')).toContain('email');
    expect(wrapper.find('input[type="password"]').prop('name')).toContain('password');
  });

  it('renders children when passed in', () => {
    props = {
      auth: {
        authIsLoading: true
      }
    };
    const wrapper = shallow((
      <SignUp {...props}>
        <RedirectLoader />
      </SignUp>
    ));
    expect(wrapper.contains(<RedirectLoader />)).toBe(true);
  });
});
