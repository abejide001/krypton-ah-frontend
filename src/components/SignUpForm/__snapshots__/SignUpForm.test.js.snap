// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Form should matches the snapshot 1`] = `
<Form
  as="form"
  onSubmit={[Function]}
>
  <FormGroup
    className="name-fields"
    widths="equal"
  >
    <div
      className="formGroup"
    >
      <FormInput
        handleChange={[Function]}
        name="firstname"
        placeholder="Firstname"
        type="text"
        value=""
      />
    </div>
    <div
      className="formGroup"
    >
      <FormInput
        handleChange={[Function]}
        name="lastname"
        placeholder="Lastname"
        type="text"
        value=""
      />
    </div>
  </FormGroup>
  <FormGroup
    widths="equal"
  >
    <FormInput
      error={false}
      handleChange={[Function]}
      name="email"
      placeholder="Email Address"
      type="email"
      value=""
    />
  </FormGroup>
  <FormGroup
    widths="equal"
  >
    <FormInput
      handleChange={[Function]}
      name="password"
      placeholder="Password"
      type="password"
      value=""
    />
  </FormGroup>
  <Button
    as="button"
    basic={true}
    fluid={true}
    huge="true"
    onClick={[Function]}
    type="submit"
  >
    Sign Up
  </Button>
</Form>
`;
